src/core/ext/filters/client_channel/resolver/google_c2p/google_c2p_resolver.cc

```diff
@@ -298,7 +298,7 @@ void GoogleCloud2ProdResolver::IPv6QueryDone(bool ipv6_supported) {
 void GoogleCloud2ProdResolver::StartXdsResolver() {
   // Construct bootstrap JSON.
   Json::Object node = {
-      {"id", "C2P"},
+      {"id", absl::StrCat("C2P-", rand())},
```

Comment:
Even though we use `rand()` in a lot of other places in core, it doesn't seem completely correct to use it given:
- AFAIK `rand()` isn't generally guaranteed to be thread safe
- it's not guaranteed to have been seeded

The backoff code [uses it's own random number generator](https://github.com/grpc/grpc/blob/5ea57ec9052bf970ac0714c3ba354c3044754a49/src/core/lib/backoff/backoff.cc#L33) which I believe solves this - this seems like the correct approach to me, any reason not to do what the backoff code is doing instead of using `rand()` ?

Reply:
See discussion in #17891.  I think `rand()` is sufficient for what we need here.

Reply:
The broken case I can imagine here binaries which aren't calling `srand` at the top of main. They would all generate the same random number in their node IDs and so defeat the purpose if the random number.

Another alternative to `rand` may be to use the absl random library, which takes care of seeding.

That said, since this is used in many other places, this is a wider discussion from this PR so doesn't need to block

Reply:
Hmm, actually, I think you're right: having every client report the same random number will kind of defeat the purpose here.  Our other uses of `rand()` inside of C-core are not visible outside of our code, so it doesn't matter as much if there's a little bit of determinism.

Unfortunately, we can't yet use the absl random library, as per https://github.com/grpc/grpc/blob/master/third_party/ABSEIL_MANUAL.md.  I've asked @veblush to look into what's blocking us from using it.  If we can get that resolved quickly, I can change this code to use that.

Reply:
For now, I've changed this to use the C++ random library.  PTAL.